hypothesis-python/src/hypothesis/vendor/pretty.py

```diff
@@ -466,8 +490,10 @@ def repr_call(
                 self.pretty(v)
                 if force_split or i + 1 < len(all_args):
                     self.text(",")
-                # Optional comments are used to annotate which-parts-matter
-                comment = comments.get(i) or comments.get(k)
+                comment = None
+                if k is not None:
+                    # Optional comments are used to annotate which-parts-matter
+                    comment = comments.get(k)
                 if comment:
```

Comment:
not confident about this, would appreciate a double-check: i is an integer so `comments.get(i)` should always be none, right? Are we calling this with an integer key in `kwargs` anywhere? Plus, am confused why `i` was being used to index into `comments` while being a loop variable. Hopefully this is just leftover code from something else which can be deleted?

Reply:
This was code to support positional arguments (which are never used by `@given()`), as part of https://github.com/HypothesisWorks/hypothesis/pull/3624.  Let's maybe add a comment block and keep it?  I'm still hoping to come back to that PR once we have `NodeTemplate` and the IR changes are done...

Reply:
gotcha - tweaked it and left it in.